src/coprocessor/dag/builder.rs

```diff
@@ -31,18 +31,18 @@ pub struct DAGBuilder;
 impl DAGBuilder {
     /// Given a list of executor descriptors and checks whether all executor descriptors can
     /// be used to build batch executors.
-    pub fn check_build_batch(exec_descriptors: &[executor::Executor]) -> Result<()> {
+    pub fn check_build_batch<S: Store + 'static>(exec_descriptors: &[executor::Executor]) -> Result<()> {
         for ed in exec_descriptors {
             match ed.get_tp() {
                 ExecType::TypeTableScan => {
                     let descriptor = ed.get_tbl_scan();
-                    BatchTableScanExecutor::check_supported(&descriptor).map_err(|e| {
+                    BatchTableScanExecutor::<S>::check_supported(&descriptor).map_err(|e| {
```

Comment:
Yes, all of these `check_supported` functions rely on a phantom generic type that does nothing. The entire pattern should change, but I didn't want to do it in this PR.

Reply:
Before this patch `BatchTableScanExecutor` accomplished this without specifying the type because `check_supported` was only implement for the concrete `FixtureStore`.

Reply:
Basically, all these functions should be free functions, not associated functions.